boards/native/include/board.h

```diff
@@ -35,6 +35,29 @@ extern "C" {
 #include "mtd_native.h"
 #endif
 
+#ifdef EUI64_PROVIDER_PARAMS_H
```

Comment:
Whaaaaaattt? I don't think it's a good idea to use a header guard here. Should this maybe be restricted to a dedicated module if it leads to so many problems?

Reply:
The problem was that before I would `#include "net/eui64.h"` to get the `eui64_t` define.
That would pull in `byteorder.h` which would define `htons()` - which conflicts with the host `htons()` definition.

Since the `EUI64_PROVIDER_` defines are only needed by `eui64_provider_params.h` and this file includes `board.h`, I found this to be a viable solution.

Reply:
Another alternative would be to move those out of `board.h` into a separate file and provide an empty fallback in `eui_provider`.
That works for `drivers/sdcard_spi` where some boards define their own `sdcard_spi_params.h`

Reply:
I don't get it... there isn't an include in this `#ifdef` so what is the inclusion problem? At any rate, this seems to me like pushing the problem to a later date, so we should fix it now! What speaks against the proposal you proposed here https://github.com/RIOT-OS/RIOT/pull/15562/files#r536829608?

Reply:
The needed include to resolve `eui64_t` is in `eui64_provider_params.h` and that includes this file after that, so it all works out…

Anyway, I agree that is is pretty revolting - #15611 will allow for a cleaner solution.

Reply:
#15611 was merged. Please rebase and adopt.

Reply:
done.